ansible/roles/operator-pipeline/templates/openshift/tasks/upload-artifact.yml

```diff
@@ -37,35 +37,92 @@ spec:
 
-        LOG_ID=$(cat output.json | jq -r ".logs._id")
-        RESULTS_ID=$(cat output.json| jq -r ".test_results._id")
+        echo "Logs URL: "
+        echo -n $LOG_URL | tee $(results.log_url.path)
+
+    - name: upload-test-results
+      image: quay.io/redhat-isv/operator-pipelines-images:latest
+      workingDir: $(workspaces.source.path)
+      env:
+        - name: PYXIS_API_KEY
+          valueFrom:
+            secretKeyRef:
+              name: $(params.pyxis_api_key_secret)
+              key: PYXIS_API_KEY
+      script: |
+        #! /usr/bin/env bash
+        PREFLIGHT_RESULTS_EXISTS="$(params.preflight_results_exists)"
+        if [ $PREFLIGHT_RESULTS_EXISTS == "true" ]; then
+          echo "Preflight results already exists - skipping"
+          echo "none" | tee $(results.result_url.path)
+          exit 0
+        fi
+
+        echo "Uploading test results"
+
+        upload-artifacts \
+          --cert-project-id "$(params.cert_project_id)" \
+          --operator-version "$(params.bundle_version)" \
+          --operator-package-name "$(params.package_name)" \
+          --certification-hash "$(params.md5sum)" \
+          --pyxis-url "$(params.pyxis_url)" \
+          --path "$(params.result_file)" \
+          --type preflight-results \
+          --output preflight-results.json \
+          --verbose
 
-        LOG_URL="$(params.pyxis_url)v1/projects/certification/id/$(params.cert_project_id)/artifacts/$LOG_ID"
+        RESULTS_ID=$(cat preflight-results.json| jq -r "._id")
         RESULT_URL="$(params.pyxis_url)v1/projects/certification/id/$(params.cert_project_id)/test-results/$RESULTS_ID"
 
-        echo "Test results URL:"
+        echo "Test result URL: "
         echo -n $RESULT_URL | tee $(results.result_url.path)
 
-        echo ""
-        echo "Logs URL: "
-        echo -n $LOG_URL | tee $(results.log_url.path)
+    - name: upload-test-artifacts
+      image: quay.io/redhat-isv/operator-pipelines-images:latest
+      workingDir: $(workspaces.source.path)
+      env:
+        - name: PYXIS_API_KEY
+          valueFrom:
+            secretKeyRef:
+              name: $(params.pyxis_api_key_secret)
+              key: PYXIS_API_KEY
+      script: |
+        #! /usr/bin/env bash
+        PREFLIGHT_RESULTS_EXISTS="$(params.preflight_results_exists)"
+        if [ $PREFLIGHT_RESULTS_EXISTS == "true" ]; then
+          exit 0
+        fi
+
+        echo "Uploading test artifacts"
+
+        upload-artifacts \
+          --cert-project-id "$(params.cert_project_id)" \
+          --operator-version "$(params.bundle_version)" \
+          --operator-package-name "$(params.package_name)" \
+          --certification-hash "$(params.md5sum)" \
+          --pyxis-url "$(params.pyxis_url)" \
+          --path "$(params.artifacts_dir)" \
+          --type preflight-artifacts \
+          --output preflight-artifacts.json \
+          --verbose
```

Comment:
No need to output/return the artifacts id/url?

Reply:
No, there can be many artifacts produced by the preflight.